packages/core/src/xhrProxy.ts

```diff
@@ -98,7 +98,7 @@ function proxyXhr() {
 
     this.addEventListener('loadend', monitor(reportXhr))
 
-    beforeSendCallbacks.forEach((callback) => callback(this._datadog_xhr))
+    beforeSendCallbacks.forEach((callback) => callback(this._datadog_xhr, this))
```

Comment:
Why not just pass `this` at that point?

Reply:
clients don't need to know that the context is attached to the xhr object through a `_datadog_xhr` property.
We could more easily change this implementation details if needed